### PR TITLE
DOC-7628 scoped AnalyticsQuery on named collections

### PR DESCRIPTION
**Pull request by osfameron**
PR for go SDK, as https://issues.couchbase.com/browse/GOCBC-1067 already resolved with pointers from @chvck thanks!

### CODE REVIEW

File: modules/devguide/examples/go/analytics-collection-scope.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	// We wait until the bucket is definitely connected and setup.
+	// For Server versions 6.5 or later if we hadn't opened a bucket then we could use cluster.WaitUntilReady here.
+	err = bucket.WaitUntilReady(5*time.Second, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	{
+		fmt.Println("handle-collection")
```
Comment (chvck) [inline modules/devguide/examples/go/analytics-collection-scope.go:33]: Oops?

**Review by chvck — APPROVED** [review]
Looks good to me.

File: modules/howtos/pages/analytics-using-sdk.adoc [hunk 2]
```diff
@@ -81,9 +81,9 @@ include::example$analytics.go[tag=options]
 
 == Handling the Response
 
-These query result may contain various sorts of data and metadata, 
-depending upon the nature of the query, 
-as you will have seen when working through our xref:6.5@server:analytics:primer-beer.adoc[introductory primer]. 
+These query result may contain various sorts of data and metadata,
```
Comment (maria-robobug) [inline modules/howtos/pages/analytics-using-sdk.adoc]: Do we need `results` here maybe?

**Review by maria-robobug — APPROVED** [review]
LGTM, just minor comment above 👍

File: modules/howtos/pages/analytics-using-sdk.adoc (hunk 2 above)
Comment (osfameron) [reply modules/howtos/pages/analytics-using-sdk.adoc]: Oops, sloppy of me to include this whitespace change in commit. I'll do a driveby commit for the plural, good spot!

Comment (osfameron) [reply modules/howtos/pages/analytics-using-sdk.adoc]: Fixed in commit below!

**Review by RichardSmedley — APPROVED** [review]
LGTM
I think we will want to rewrite the intro for 2.3, removing all of the Data Platform and 5.5DP bits,
but fine for now  :+1:

**Pull request merged**